const ApiError = require("../../utils/apiError");

class ShowProductService {
  constructor(ProductRepository) {
    this.productRepository = ProductRepository;
  }

  async execute(id) {
    let product = await this.productRepository.findById(id);

    if (!product) {
      throw new ApiError(404, "Produto n√£o encontrado");
    }
    return product;
  }
}

module.exports = ShowProductService;
